azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CreateResourcesThenCreateNodes.java

```diff
@@ -117,24 +114,22 @@ protected CreateResourceGroupThenCreateNodes(
 
       // This sill create the resource group if it does not exist
       String location = template.getLocation().getId();
-      ResourceGroup resourceGroup = resourceGroupMap.getUnchecked(location);
-      String azureGroupName = resourceGroup.name();
 
-      getOrCreateVirtualNetworkWithSubnet(location, options, azureGroupName);
-      configureSecurityGroupForOptions(group, azureGroupName, template.getLocation(), options);
+      createResourceGroupIfNeeded(group, location, options);
```

Comment:
I think you are right although I'd probably prefer to read something like
```
...
      if (options.getResourceGroup() != null) {
         group = options.getResourceGroup();
      }


   private ResourceGroup getOrCreateResourceGroup(String group, String location) {
      if (group != null) {
      logger.debug(">> using resource group [%s]", group);
      ResourceGroup rg = api.getResourceGroupApi().get(group);
      if (rg != null)  return rg;

      logger.debug(">> resource group [%s] does not exist. Creating!", group);
      return api.getResourceGroupApi().create(group, location,
            ImmutableMap.of("description", "jclouds default resource group"));
      }
   }
```
and then set the resourceGroup.name on the options object where needed.

but maybe it is not significantly better. Again, just a minor comment :)